pkg/app/server.go

```diff
@@ -274,6 +270,23 @@ func RunKubeStateMetrics(ctx context.Context, opts *options.Options) error {
 
 	tlsConfig := opts.TLSConfig
 
+	// A nil CRS config implies that we need to hold off on all CRS operations.
+	if config != nil {
+		// This starts a goroutine that will watch for any new GVKs to extract from CRDs.
+		err = discovererInstance.StartDiscovery(ctx, kubeConfig)
```

Comment:
since discovererInstance is only used here, it would be better to initialize it here instead passing it down two calls as a parameter